### PR TITLE
build(eslint): Add eslint package and fix errors

### PR DESCRIPTION
**Pull request by pmareke**
EX-4211

Added `@empathyco/eslint-plugin-x` dependency and fix lint errors.

### CODE REVIEW

File: packages/deep-merge/package.json [hunk 1]
```diff
@@ -34,12 +34,13 @@
     "build:local": "webpack --mode production && npm pack",
     "clean": "rimraf dist && rimraf coverage",
     "format": "prettier --write \"{src,__tests__}/**/*.ts\" --single-quote --trailing-comma es5",
```
Comment (javieri-empathy) [inline packages/deep-merge/package.json:36]: Use prettier preset too

File: packages/deep-merge/src/deep-merge.ts [hunk 2]
```diff
@@ -64,11 +95,25 @@ function cloneObjectProperties(target: any, [key, value]: any): any {
...
+function isObject(obj: any): obj is Record<string, unknown> {
+  return obj && typeof obj === "object" && !Array.isArray(obj);
 }
 
-function mergeObject(target: any, [key, value]: any): any {
+/**
+ * A function which merges two sources.
+ *
+ * @param target - The target object to clone in.
+ * @param source - The source objet to clone.
```
Comment (javieri-empathy) [inline packages/deep-merge/src/deep-merge.ts:113]: This is not the `source` object, but an entry (key, value) to clone into the `target` object

File: packages/deep-merge/src/deep-merge.ts [hunk 3]
```diff
@@ -35,22 +45,43 @@ export function replaceBehaviour<T extends object>(obj: T): T {
...
   } else {
     return target || {};
   }
 }
 
+/**
+ * A function which clones the properties of two object.
+ *
+ * @param target - The target object to clone in.
+ * @param source - The source objet to clone.
```
Comment (javieri-empathy) [inline packages/deep-merge/src/deep-merge.ts:81]: This is not the source object, but an entry (key, value) to clone into the target object

File: packages/deep-merge/src/deep-merge.ts [hunk 4]
```diff
@@ -35,22 +45,43 @@ export function replaceBehaviour<T extends object>(obj: T): T {
...
+ *
+ * @returns The union of the target and source.
```
Comment (javieri-empathy) [inline packages/deep-merge/src/deep-merge.ts:83]: I would suggest changing it to something like:
> @returns The `target` object updated with the `entry` parameter.

File: packages/deep-merge/.eslintrc [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": ["plugin:@empathyco/x/all"]
```
Comment (javieri-empathy) [inline packages/deep-merge/.eslintrc:2]: The `deep-merge.spec.ts` file is not read by eslint due to the `tsconfig.json`. We can fix it by simply moving it to `src/__tests__` like we do in x-components

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: packages/deep-merge/package.json [hunk 6]
```diff
@@ -33,13 +33,14 @@
     "build": "tsc && tsc --project tsconfig.esm.json",
     "build:local": "webpack --mode production && npm pack",
     "clean": "rimraf dist && rimraf coverage",
-    "format": "prettier --write \"{src,__tests__}/**/*.ts\" --single-quote --trailing-comma es5",
-    "lint": "",
+    "format": "prettier --write \"src/**/*.ts\" --single-quote --trailing-comma es5",
```
Comment (javieri-empathy) [inline packages/deep-merge/package.json:36]: Remove the inlined properties (`--single-quote`, `--trailing-comma`, `es5`)

File: packages/deep-merge/src/deep-merge.ts [hunk 7]
```diff
@@ -1,11 +1,18 @@
+enum Behaviour {
+  Replace = 'replace',
+  DeepMerge = 'deep-merge',
```
Comment (javieri-empathy) [inline packages/deep-merge/src/deep-merge.ts:3]: Thanks Github

File: packages/deep-merge/src/tests/deep-merge.spec.ts [hunk 8]
```diff
@@ -1,4 +1,4 @@
-import { deepMerge, deepMergeBehaviour, replaceBehaviour } from '../src/deep-merge';
+import { deepMerge, deepMergeBehaviour, replaceBehaviour } from '../deep-merge';
```
Comment (javieri-empathy) [inline packages/deep-merge/src/tests/deep-merge.spec.ts:1]: Move the file to `src/__tests__`, which is x-components standard.

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

**Review by javieri-empathy — APPROVED** [review]

File: packages/deep-merge/tsconfig.eslint.json [hunk 9]
```diff
@@ -0,0 +1,4 @@
+{
+  "extends": "./tsconfig.json",
+  "exclude": []
```
Comment (LuisMartinez15) [inline packages/deep-merge/tsconfig.eslint.json:3]: I don't know why we keep adding this `"exclude"` property if we are setting it to an empty array. Am I missing something? 😅

**Review by LuisMartinez15 — APPROVED** [review]

**Pull request merged**